# ET-DDF instance configuration

# all connections in network -> read as connections[0] = agent 0's connections
# connections: [[1,2,3],[0],[0],[0]]
# connections: [[1,2],[0],[0,3],[2]]
connections: [[4,5],[3],[3,6],[5]]

# update rate for et-ddf code
agent_update_rate: 5 # [Hz]

# message statistics update rate
msg_stats_rate: 1 # [hz]

# event-triggering parameters
delta: 1.5    # measurement-sharing Kalman innovation threshold param
tau: 7         # covariance intersection uncertainty per state threshold

# probability of dropping messages
comm_drop_prob: 0.0

# adaptive CI thresholding params
use_adaptive_tau: True
epsilon_1: 0.1          # adaptive threshold heuristic param -- weight for change from old value
epsilon_2: 0.01         # adaptive threshold heuristic param -- weight for balancing w/ neighbors

# starting position
start_pos:
    x: 0.0
    y: 0.0
    z: 0.0
    phi: 0.0
    theta: 0.0
    psi: 0.0

# dynamics model used in filters
dynamics: 
    lin_ncv_3d:
        coplanar: False

# sensor models used in filters
sensors:
    gps:
        agents: [3] # read as only agent 3 gets GPS
        noise: [[1,0],[0,1]]
    lin_rel:
        agents: [3,4,5,6]
        noise: [[3,0],[0,3]]
